kubernetes_platform/python/kfp/kubernetes/config_map.py

```diff
@@ -79,12 +83,18 @@ def use_config_map_as_volume(
     msg = common.get_existing_kubernetes_config_as_message(task)
 
     config_map_as_vol = pb.ConfigMapAsVolume(
-        config_map_name=config_map_name,
         mount_path=mount_path,
         optional=optional,
     )
-    msg.config_map_as_volume.append(config_map_as_vol)
 
+    config_map_name_parameter = common.parse_k8s_parameter_input(config_map_name, task)
+    config_map_as_vol.config_name_parameter.CopyFrom(config_map_name_parameter)
```

Comment:
This is safe because if this is submitted against an old API server, this field will be ignored, right?

Reply:
No this is not safe due to un-marshalling of the executor config which poses new problems, the current thinking is to not deprecate the older fields and support both. For example for `str` inputs use `config_map_name`, for inputs use `config_name_parameter`.

This won't work for pvc mounts unfortunately, since the inputs are already of type parameter channel, so there's no way to distinguish between old/new apis, so this would probably require deprecation.